### DISCUSSION TITLE
Can Hy read and rewrite python code?

### DISCUSSION THREAD

**Discussion by Kurvivor19** [hylang/hy#2339] [Q&A] [answered]
Hy can be used as a generator for python code due to built-in compilation into python. Canit also be used for manipulating existing one? 
What i am envisioning is domething like reading *.py file and parsing it into syntax tree that can be modified before evaluating it or writing back. Is there capability in Hy for that?

**Comment by Kodiologist** [comment]
No, see #1721.

**Comment by Kodiologist** [comment, marked as answer]
There's now an official [py2hy](https://github.com/hylang/py2hy). Also, in hindsight, I seem to have misread the question. The answer was always yes: you can use `ast.parse` and `ast.unparse` just as you would in Python.
